api/txspool.go

```diff
@@ -47,6 +47,27 @@ func (a *API) postAtomicPool(c *gin.Context) {
 		retBadReq(errors.New(ErrSingleTxInAtomicEndpoint), c)
 		return
 	}
+	// set the Rq fields
+	for _, tx1 := range receivedTxs {
+		for i, tx2 := range receivedTxs {
+			if tx1.RqTxID == tx2.TxID {
+				tx1.RqFromIdx = tx2.FromIdx
```

Comment:
I'd suggest to not use this pattern when writing to the items you're iterating over, since some times there is unexpected pointer related behavior.

Instead I'd do something like:

`receivedTxs[i].RqFromIdx = receivedTxs[j].FromIdx`